beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -341,6 +347,19 @@ func (vs *Server) proposeGenericBeaconBlock(ctx context.Context, blk interfaces.
 		return nil, errors.Wrap(err, "could not unblind builder block")
 	}
 
+	if validation == types.Consensus {
+		if err = vs.validateConsensus(ctx, blk); err != nil {
+			return nil, errors.Wrap(err, types.ErrConsensusValidationFailed.Error())
+		}
+	} else if validation == types.ConsensusAndEquivocation {
+		if err = vs.validateConsensus(ctx, blk); err != nil {
+			return nil, errors.Wrap(err, types.ErrConsensusValidationFailed.Error())
+		}
+		if err = vs.validateEquivocation(blk.Block()); err != nil {
+			return nil, errors.Wrap(err, types.ErrEquivocationValidationFailed.Error())
+		}
+	}
+
```

Comment:
I would prefer these checks to be outside of `v1alpha1` pkg. It should be in the beacon-api pkg. 
1.) It's less risky to current validator code
2.) v1alpha1 will be deprecated eventually then we shouldn't have to move these checks again

Reply:
From the spec:
> `consensus_and_equivocation`: the same as `consensus`, with an extra equivocation check immediately before the block is broadcast 

Broadcast is done here, so it's hard to move this check anywhere else. Also I believe the function will stay as it is even when we deprecate v1alpha1. It will be just no longer a method on the v1alpha1 server, but rather a regular function either in Beacon APIs or somewhere else.

Reply:
Why not break it up and call Broadcast directly like we did here: https://github.com/prysmaticlabs/prysm/pull/12335/files#diff-50db3f4907c1c24a971fc3a1ef6264d8386f7ec2b42cf60928238b6dfc94d058R1044

I think we should delete `v1alpha1` eventually so avoid adding more dependency to it will be nice.